---
title: "Final Report: The Conditioned Murine Alveolar Immune Response"
author: "Anqi Dai"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    code_folding: hide
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
library(tidyverse)
library(knitr)
library(ggpubr)
```

The Mizgerd lab recently performed gene expression profiling with microarrays to understand and characterize the role of AMs in a mouse model of pulmonary infection. We sought to test hypothesis that experienced mice with previous *S. pneumoniae* exposure may have an AM response that better models the human response than naive mice, which exhibit significant differences from the human immune response.

This analysis compared the gene expression results from the human AM study presented in the [Bewley paper](https://www.atsjournals.org/doi/full/10.1164/rccm.201705-0903OC?url_ver=Z39.88-2003&rfr_id=ori%3Arid%3Acrossref.org&rfr_dat=cr_pub%3Dpubmed) with the transcriptional responses observed in naive exposed vs control and experience exposed vs control mouse experiments. 

## 1. Bewley data processing

Only the expression from healthy human subjects were analyzed, where 3 exposed vs 3 control AM expression datasets underwent QC and differential expression analysis as described in the supplemental material of the original manuscript. 

### 1.1. QC and preprocessing

#### 1.1.1 QC of the raw data

For quality control of the raw data, the median relative log expression (RLE) of each sample was computed. Results are shown in the below barplot. The fact that all samples have a median RLE close to zero suggests high quality.

```{r out.width=400}
knitr::include_graphics('../figs/median_rle.jpg')
```

The median normalized unscaled standard error (NUSE) for each sample was calculated as well. The samples with NUSE > 1 may be poor quality. The resulted barplot shows the quality is high.

```{r out.width=400}
knitr::include_graphics('../figs/median_NUSE.jpg')
```

#### 1.1.2. Preprocessing

* Raw data was analyzed to perform background correction using GC Robust Multi-array Average, and quantile normalization. 
* Probes whose expression was in the 20th lowest centile were removed. 

### 1.2. Differential expression

* Differentially expressed genes were identified using Limma to calculate Moderated t tests and considered significant if adjusted p value < 0.05 following Benjamini-Hochberg correction. 

[TABLE: DE statistics for all the probesets remained after filtering](../output/differential_expression_statistics_healthy_samples_Bewley_ALL.csv)

[TABLE: DE statistics for significant results only](../output/differential_expression_statistics_healthy_samples_Bewley_SIG.csv)

Since we also obtained the original author's data on the differential expression statistics. Below is a comparison of the their results and ours.

```{r out.width=400}
knitr::include_graphics('../figs/Overlap between Bewley and I sig result.png')
```

The above Venn diagram indicates that all of the significant genes from authors' analysis are included in our results. 

And to inspect whether the expression are concordant between us, the Spearman correlaton is computed on both the log2 fold change and the average expression of the significant genes. The results are listed below:

Comparison | Correlation
---------- | -----------
Bewley_log2FC VS My_log2FC | 1
Bewley_aveExpr VS My_aveExpr | 1

The sample estimates rho are both 1 for the two comparisons, proving that not only are the genes identified as Differentially Expressed Genes (DEG) are agreeable between the authors' results and ours, but also the statistics are comparable.



### 1.3. GSEA

* Only keep the genes that are homologous in human and in mouse.
    + Identify the Entrez gene ID for the probesets in the human data DE result.
    + For multiple probeset that correspond to the same Entrez ID, only keep the one that has the maximum absolute fold change.
    + For the mouse results provided by BU Microarray Core, since it already includes the homologous human gene information if any, filter out the mouse probes that do not have a homologous human gene.
    + For the situation where multiple mouse probes correspond to the same human gene, simply remove those.
    + Join the mouse and human table together, and a table with mouse and human data having one-to-one relationship is produced.


* Use the t statistics from the differential expression results to do a Gene Set Enrichment Analysis (GSEA) employing the fgsea Bioconductor package with the MsigDB gene set database version 6.2, looking into only the c2.cp (canonical pathways), c5.all (GO gene sets) and c7.all (immunological signatures) enrichment.

[TABLE: All pathway enrichment statistics for Bewley human data](../output/HDE_GSEA_results.csv)


## 2. DEGs comparisons

Three DE analyses were used for the following comparisons:

* HDE: Healthy human exposed AMs vs control AMs
* EDE: Experienced mouse exposed AMs vs control AMs
* NDE: Naive Mouse exposed AMs vs control AMs

[TABLE: Homolog-mapped statistics including log2 fold change,FDR and t statistics for the HDE, NDE and EDE](../output/homolog_mapped_statistics_HDE_NDE_EDE.csv)

### 2.1. Spearman correlation

Spearman correlation of homolog-mapped log2 fold changes for HDE vs EDE and HDE vs NDE:

Comparison | Correlation
---------- | -----------
HDE VS EDE | -0.000209333
HDE VS NDE | -0.05137032

The number suggests there is no correlation between either of the two.

### 2.2. Scatter plot

Scatter plot of HDE VS EDE log2 fold change:

```{r out.width=400}
knitr::include_graphics('../figs/HDE VS EDE log2 fold change.jpg')
```

Scatter plot of HDE VS NDE log2 fold change:

```{r out.width=400}
knitr::include_graphics('../figs/HDE VS NDE log2 fold change.jpg')
```

Still, both of the plots seem to have points randomly located, suggesting no correlation found.

### 2.3. Table and Venn diagrams

Tables and Venn diagrams of significant DE gene overlap for HDE, EDE, and NDE for FDR levels 0.1, 0.05, and 0.01.

Group | FDR01 | FDR005 | FDR001
----- | ----- | ------ | ------
HDE NDE overlap | 1442 | 840 | 184
HDE EDE overlap | 839 | 439 | 90

```{r out.width=400}
knitr::include_graphics('../figs/Overlap between HDE, NDE and EDE (FDR < 0.1).png')
```

```{r out.width=400}
knitr::include_graphics('../figs/Overlap between HDE, NDE and EDE (FDR < 0.05).png')
```

```{r out.width=400}
knitr::include_graphics('../figs/Overlap between HDE, NDE and EDE (FDR < 0.01).png')
```

Contrary to what was expected, the Human DE genes had more overlap with naive mouse group than experienced group in all three significance levels.

## 3. GSEA comparisons

### 3.1 Table and Venn diagrams

Following the same steps as described in 1.3., the GSEA is done for EDE and NDE.

Table and Venn diagrams of significant enriched gene sets from GSEA analyses for HDE, EDE, and NDE for FDR levels 0.1, 0.05, 0.01, with a Venn diagram for each level.

Group | FDR01 | FDR005 | FDR001
----- | ----- | ------ | ------
HDE NDE overlap | 2372 | 1747 | 792
HDE EDE overlap | 2363 | 1753 | 822

```{r out.width=400}
knitr::include_graphics('../figs/Overlap enriched pathway between HDE, NDE and EDE (FDR < 0.1).png')
```

```{r out.width=400}
knitr::include_graphics('../figs/Overlap enriched pathway between HDE, NDE and EDE (FDR < 0.05).png')
```

```{r out.width=400}
knitr::include_graphics('../figs/Overlap enriched pathway between HDE, NDE and EDE (FDR < 0.01).png')
```

The comparison demonstrates that the three GSEA results have tremendous overlap, but the overlap between HDE and EDE is not necessarily higher than HDE and NDE.

### 3.2 Nature Figure 4b

GSEA statistics for gene sets corresponding to the top and bottom 400 genes in EDE and NDE computed against the HDE t statistic dataset as a table and leading edge plots as in [Beura et al](https://www.nature.com/articles/nature17655) Figure 4b.

Detailed steps:

* Filter to select the significant genes in NDE (FDR < 0.05)
* Sort the t statistic in descending order and retrieve the top and bottom 400 entrez IDs
* Repeat the above 2 steps for EDE
* Make a gene set file using the above 4 list of genes
* Check the enrichment of those 4 gene sets from mouse in human using the human t statistic.

The resulted plots are shown below. EDE_up represents the gene set that consists of top 400 upregulated genes from EDE. EDE_down the 400 most downregulated genes. Similar for the NDE_up and NDE_down.


```{r out.width='50%'}
knitr::include_graphics(c('../figs/EDE_up_enrichment.jpg','../figs/EDE_down_enrichment.jpg' ))
knitr::include_graphics(c('../figs/NDE_up_enrichment.jpg','../figs/NDE_down_enrichment.jpg' ))
```

The plots point out that there is generally a higher enrichment of mouse upregulated genes in both NDE and EDE compared to downregulated genes. And the enrichment of upregulated genes from EDE is slightly higher than from NDE.

## 4. Conclusions

To conclude, this analysis almost fully reproduced the steps in Bewley paper on the DE of healthy human samples, but the comparison between human and mouse is not in support of our hypothesis. The data is not indicative of experienced mice with previous *S. pneumoniae* exposure better model the human response than naive mice.



